ThunkLibs/libX11/libX11_Host.cpp

```diff
@@ -25,103 +25,301 @@ desc: Handles callbacks and varargs
 #include <utility>
 
 #include "ldr_ptrs.inl"
+#ifdef _M_ARM_64
```

Comment:
Are you planning to implement a proper x86 code path?

Maybe we can just implement a generic (if less performant) fallback template for non-ARM architectures, with the same logic as before but more cases.

Reply:
I don't have the heart at the moment to write an x86 specific code path at the moment for this. For now it hits the C path and I added the 8 case variadic version which would be required.
Main purpose for the ARM side is that it now covers the variadic path regardless of number of arguments passed in rather than any sort of performance optimization. Slightly elevated performance is a by-product but not the reason its ASM.

Reply:
Sounds fair. If the final version of this PR moves back to using a InternalVariadic helper function, would be nice to adapt the non-arm bits to also use this function (implemented in C for `count<8` on those platforms), so the host endpoints don't need to be specialized per-arch.

Reply:
Moved back to a function call for the default switch statement case instead. We can take the minor perf hit here for now.